File: rb/features.rb
# EvilInsultGenerator SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module EvilInsultGeneratorFeatures
  def self.make_feature(name)
    case name
    when "base"
      EvilInsultGeneratorBaseFeature.new
    when "test"
      EvilInsultGeneratorTestFeature.new
    else
      EvilInsultGeneratorBaseFeature.new
    end
  end
end

File: py/evilinsultgenerator_sdk/features.py
# EvilInsultGenerator SDK feature factory

from evilinsultgenerator_sdk.feature.base_feature import EvilInsultGeneratorBaseFeature
from evilinsultgenerator_sdk.feature.test_feature import EvilInsultGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EvilInsultGeneratorBaseFeature(),
        "test": lambda: EvilInsultGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
